README.md

```diff
@@ -1,634 +1,1828 @@
+        - `assetBeacon` == `assetBeacon` from corresponding Offer UTxO
+        - `loanPrinciple` == `loanPrinciple` from corresponding Offer UTxO
+        - `compoundFrequency` == `compoundFrequency` from corresponding Offer UTxO
+        - `lastCompounding` == invalid-before of this transaction (as POSIXTime)
+        - `loanTerm` == `loanTerm` from corresponding Offer UTxO
+        - `loanInterest` == `loanInterest` from corresponding Offer UTxO
+        - `minPayment` == `minPayment` from corresponding Offer UTxO
+        - `penalty` == `penalty` from corresponding Offer UTxO
+        - `collateralization` == `collateralization` from corresponding Offer UTxO
+        - `collateralIsSwappable` == `collateralIsSwappable` from corresponding Offer UTxO
+        - `claimExpiration` == invalid-before of this transaction (as POSIXTime) + `loanTerm` +
+        `claimPeriod`
+        - `loanExpiration` == invalid-before of this transaction (as POSIXTime) + `loanTerm`
+        - `loanOutstanding` == `loanPrinciple` * (1 + `loanInterest`)
+        - `totalEpochPayments` == 0
+        - `loanId` == `sha2_256( offer_utxo_tx_hash ++ offer_utxo_output_index )`
+    - It must have enough relative collateral to equal the `loanPrinciple`.
+    - It must have exactly 1 Borrower ID with the token name matching `borrowerId`, 1 Active beacon
+    with the token name "Active", 1 loan asset beacon with the token name matching `assetBeacon`,
+    and 1 Loan ID beacon with the token name matching `loanId`.
+- Every offer must have a corresponding lender payment output:
+    - It must be locked at the `lenderAddress` in the corresponding `ActiveDatum`
+    - It must contain the `offerDeposit` from the corresponding `OfferDatum` and exactly 1 Loan ID
+    beacon with the token name matching `loanId` in the corresponding `ActiveDautm`
+    - A `PaymentDatum` where the `CurrencySymbol` is the active smart contract policy id and the
+    `TokenName` is the `loanId` from the corresponding `ActiveDatum`
+- The negotiation smart contract must burn *all* negotiation beacons attached to the Ask and Offer
+UTxOs.
+- The active smart contract must mint exactly:
+    - 1 Active beacon for every Offer UTxO among the inputs.
+    - 1 Borrower ID beacon for every Offer UTxO among the inputs.
+    - 2 Loan ID beacons for every Offer UTxO among the inputs (each Offer UTxO gets their own unique
+    Loan ID pair).
+    - 1 loan asset beacon for every Offer UTxO accepted that requires this asset beacon.
+- The staking credential of the loan address where all inputs come from must approve the
+transaction.
+
+*Valid Offers and Asks are those containing the proper beacons.*
+
+The loan's start time is set to the invalid-before bound of the acceptance transaction's validity
+interval. If any offers have an expiration set, then the invalid-hereafter bound will also be needed
+to prove the offer has not expired. If no offers have expirations set, the invalid-hereafter bound
+can be dropped.
```

Comment:
```suggestion
The loan's start time is set to the `invalid-before` bound of the acceptance transaction's validity
interval. If any offers have an expiration set, then the `invalid-hereafter` bound will also be needed
to prove the offer has not expired. If no offers have expirations set, the `invalid-hereafter` bound
can be dropped.
```
... to clarify that these are fixed parameters of the transaction? (used in other places as well).  Maybe not needed if audience is only the Cardano community.